lib/pure/strmisc.nim

```diff
@@ -85,6 +86,80 @@ proc rpartition*(s: string, sep: string): (string, string, string)
 
   return partition(s, sep, right = true)
 
+
+func parseFloatThousandSep*(str: openArray[char]; sep = ','; decimalDot = '.'): float {.since: (1, 5).} =
+  ## Convenience func for `parseFloat` which allows for thousand separators,
+  ## this is designed to parse floats as found in the wild formatted for humans.
+  ##
+  ## The following assumptions and requirements must be met:
+  ## - `str` must not be empty.
+  ## - `str` must be stripped of trailing and leading whitespaces.
+  ## - `sep` must not be `'-'`.
+  ## - `decimalDot` must not be `'-'` nor `' '`.
+  ## - `sep` and `decimalDot` must be different.
+  ## - No separator before a digit.
+  ## - There has to be 3 digits between successive separators.
+  ## - There has to be 3 digits between the last separator and the decimal dot.
+  ## - No separator after decimal dot.
+  ## - No duplicate separators.
+  ## - Floats without separator allowed.
+  ##
+  ## See also:
+  ## * `parseFloat <strutils.html#parseFloat,string>`_
+  runnableExamples:
+    doAssert parseFloatThousandSep("-1,000") == -1000.0
+    doAssert parseFloatThousandSep("10,000.000") == 10000.0
+    doAssert parseFloatThousandSep("1,000,000.000") == 1000000.0
+    doAssert parseFloatThousandSep("10,000,000.000") == 10000000.0
+    doAssert parseFloatThousandSep("10.000,0", '.', ',') == 10000.0
+    doAssert parseFloatThousandSep("1'000'000,000", '\'', ',') == 1000000.0
+    doAssert parseFloatThousandSep("1000000") == 1000000.0
+    ## You can omit `sep`, but then all subsequent `sep` to the left must also be omitted:
+    doAssert parseFloatThousandSep("1000,000") == 1000000.0
+
+  assert sep != '-' and decimalDot notin {'-', ' '} and sep != decimalDot
+
+  proc raiseError(i: int; c: char; s: openArray[char]) {.noinline, noreturn.} =
+    raise newException(ValueError,
+      "Invalid float containing thousand separators, invalid char $1 at index $2 for input $3" %
+      [c.repr, $i, s.repr])
```

Comment:
TODO for future PR:
this will show weird looking errors:
`input ['1', '2', ',', '1', '.'] [ValueError]`
instead of:
`input "12,1." [ValueError]`

I sent out https://github.com/nim-lang/Nim/pull/15951 ; when this is merged we can revisit this using `add`